Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerMentionCellView_Tests.swift

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import StreamChatTestTools
+@testable import StreamChatUI
+import XCTest
+
+class ChatMessageComposerMentionCellView_Tests: XCTestCase {
+    /// Default reference width for the cell. Not related to any screen size.
+    private static var defaultCellWidth: CGFloat = 300
+
+    private var chatUserOffline: _ChatUser<NoExtraData>!
+    private var chatUserOnline: _ChatUser<NoExtraData>!
+    private var chatUserNoName: _ChatUser<NoExtraData>!
+
+    override func setUp() {
+        super.setUp()
+        chatUserOffline = .mock(
+            id: "darkside37",
+            name: "Mr Vader",
+            imageURL: TestImages.vader.url,
+            isOnline: false
+        )
+
+        chatUserOnline = .mock(
+            id: "darkside37",
+            name: "Mr Vader",
+            imageURL: TestImages.vader.url,
+            isOnline: true
+        )
+
+        chatUserNoName = .mock(
+            id: "yoda",
+            imageURL: TestImages.yoda.url,
+            isOnline: true
+        )
+    }
+
+    func test_emptyAppearance() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        AssertSnapshot(view)
+    }
+
+    func test_defaultAppearance() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        view.content = chatUserOnline
+        AssertSnapshot(view, suffix: "online indicator visible")
+
+        view.content = chatUserOffline
+        AssertSnapshot(view)
+
+        view.content = chatUserNoName
+        AssertSnapshot(view, suffix: "user name not set")
+    }
+
+    func test_appearanceCustomization_usingUIConfig() {
+        class RectIndicator: UIView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        class CustomAvatarView: _ChatChannelAvatarView<NoExtraData> {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        var config = UIConfig()
+        config.channelList.channelListItemSubviews.onlineIndicator = RectIndicator.self
+        config.messageComposer.mentionAvatarView = CustomAvatarView.self
```

Comment:
If this is related to the message composer why are we using channeList views? 🤔  Maybe the onlineIndicator should be in a more generic place?

Reply:
Hmm actually very good comment. However please create ticket for this, I figured out that the structure of `UIConfig` is object to change 😅  I mean with this change we would probably need to change the structure a bit, because it's quite different from `ChannelList`...